cli/containerdutils/hostpaths.go

```diff
@@ -0,0 +1,294 @@
+		}
+		status, err := task.Wait(ctx)
+		if err != nil {
+			logrus.Fatalf("Failed to wait for container task: %s", err)
+		}
+		<-status
+		err = task.Kill(ctx, SIGKILL)
+		if err != nil && !strings.Contains(err.Error(), "not found") {
+			return fmt.Errorf("failed to kill task: %s", err)
+		}
+		_, err = task.Delete(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to delete task: %s", err)
+		}
+		return container.Delete(ctx, containerd.WithSnapshotCleanup)
+	}
+	return fmt.Errorf("Failed to create path %s", dirname)
+}
+
+// VerifyDockerConfig ensures the specified docker config file exists on the host and returns the rootDir
+func VerifyDockerConfig(ctx context.Context, client WrappedContainerdClient, configFile string, image containerd.Image) (string, error) {
+
+	// First make sure the dir exists on the host
+	configDir := path.Dir(configFile)
+	err := VerifyHostDirExists(ctx, client, configDir, image)
+	if err != nil {
+		return "", fmt.Errorf("failed to verify host dir %s: %s", configDir, err)
+	}
+
+	// TODO - consider refining this to be a generic utility to create a file on the host
+	// with the file content passed in as a string
+	testerSpec := specs.Spec{
+		Root: &specs.Root{
+			Path: "rootfs",
+		},
+		Process: &specs.Process{
+			Cwd: "/",
+			Args: []string{
+				"sh", "-c",
+				fmt.Sprintf("[ -f \"%s\" ] || echo \"{}\" > \"%s\" ; cat \"%s\"", configFile, configFile, configFile),
+			},
+			User: specs.User{
+				UID: 0,
+				GID: 0,
+			},
+			Env: []string{
+				"PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin",
+			},
+			NoNewPrivileges: false,
+		},
+	}
+	name := stringid.GenerateRandomID()[:20]
+	cOpts := []containerd.NewContainerOpts{
+		containerd.WithNewSnapshot(name, image),
+		containerd.WithImage(image),
+		containerd.WithSpec(&testerSpec,
+			WithHostMounts,
+			WithAllCapabilities,
+			WithHostPropagation,
+		),
+	}
+	testerSpec.Mounts = []specs.Mount{
+		{
+			Options: []string{
+				"rbind",
+				"rw",
+				"rshared",
+			},
+			Source:      path.Dir(configFile),
+			Destination: path.Dir(configFile),
+			Type:        "bind",
+		},
+	}
+	container, err := client.NewContainer(
+		ctx,
+		name,
+		cOpts...,
+	)
+	if err != nil {
+		return "", fmt.Errorf("Failed to verify config file %s: %s", configFile, err)
+	}
+	// Capture the output
+	buf := new(bytes.Buffer)
+	task, err := container.NewTask(ctx, defaultNewCreator(withByteBuffers(buf)))
+	if err != nil {
+		return "", fmt.Errorf("Failed to verify %s: %s", configFile, err)
+	}
+	err = task.Start(ctx)
+	if err != nil {
+		return "", fmt.Errorf("Failed to start container task: %s", err)
+	}
+	status, err := task.Wait(ctx)
+	if err != nil {
+		return "", fmt.Errorf("Failed to wait for container task: %s", err)
+	}
+	<-status
+	err = task.Kill(ctx, SIGKILL)
+	if err != nil {
+		logrus.Debugf("Failed to kill task: %s", err)
+	}
+	_, err = task.Delete(ctx)
+	if err != nil {
+		logrus.Debugf("Failed to delete task: %s", err)
```

Comment:
almost silently ignoring these errors :thinking:  shouldn't we at least print some warnings ?